src/main/scala/io/citrine/lolo/trees/FeaturePath.scala

```diff
@@ -0,0 +1,96 @@
+package io.citrine.lolo.trees
+
+/**
+  * Description of a feature's role in computing TreeSHAP
+  *
+  * @param featureIndex index of the feature this node describes.
+  * @param zeroFraction fraction of paths flowing through this node with this feature excluded.
+  * @param oneFraction fraction of one paths flowing through this node with this feature included.
+  * @param pathWeight summation weight for this path, taking into account combinatorial factor.
+  */
+case class FeatureNode(
+                  var featureIndex: Int,
```

Comment:
`var` in case classes is a big anti-pattern.  Case classes should almost always be immutable.  This mutability would be very confusing to future developers, possibly leading to bugs.

Reply:
Thanks. I've switched to regular classes where the algorithm uses mutability. It would be possible to refactor to use immutable case classes, but that would add complication and I'd like to get a MVP shipped.